radixdlt/src/main/java/com/radixdlt/consensus/validators/ValidatorSet.java

```diff
@@ -57,7 +70,24 @@ public static ValidatorSet from(Collection<Validator> validators) {
 	 * @return An initial validation state with no signatures
 	 */
 	public ValidationState newValidationState(Hash anchor) {
-		return ValidationState.forValidatorSet(anchor, validators.keySet());
+		return ValidationState.forValidatorSet(anchor, this);
+	}
+
+	public boolean containsKey(ECPublicKey key) {
+		return validators.containsKey(key);
+	}
+
+	public UInt256 getPower(Set<ECPublicKey> signedKeys) {
+		return signedKeys.stream()
+			.map(validators::get)
+			.map(Validator::getPower)
```

Comment:
Should filter nulls after validators::get, as there is no enforcing that all the signedKeys are required to be in the validator set from the interface.

This perhaps could be specified in the Javadoc for this method, but there is none.